pkg/cli/cli.go

```diff
@@ -663,11 +662,17 @@ func runConnectionAgentCommand(flags *ConnectFlags, c *command, args []string, o
 		return nil
 	}
 
-	// Signal the caller that the agent is moving to the background.
-	// Ideally, this should close the command's streams, but those are not closeable.
-	os.Stdin.Close()
-	os.Stdout.Close()
-	os.Stderr.Close()
+	// Signal the caller that the agent is moving to the background by closing
+	// the command's standard IO channels.
+	if cin, closeable := c.InOrStdin().(io.Closer); closeable {
```

Comment:
nit: it's fine to use `ok` here to keep it consistent with the use of the `ok idiom` in the code base.

Reply:
Done